### PR TITLE
(1.3.2) Documentation import

### PR DESCRIPTION
**Pull request by IvanSavenko**
As discussed on Slack, moving all vcmi documentation to a single location: repository.

Reasoning:
- Single location, compared to currently scattered docs across website wiki, github wiki and github repo
- Can be updated as part of PR's, to avoid docs getting stale when author of changes forgets to update docs after release
- Implicitly versioned alongside with vcmi - you can check master branch for docs related to latest release, develop branch - for next version docs, or pick tag for docs related to specific release.

Conversion from mediawiki/latex format to Github markdown done using https://pandoc.org/try/

Documentation import progress:
- [x] Convert manual to markdown
- [x] Import Github wiki data
- [x] Import website wiki

Documentation review progress:
- [x] Look up for any missing / undocumented content
- [x] Review file names / locations
- [x] Review & restore links between files (got lost during import)
- [x] Review & update players documentation 
- [x] Review & update modders documentation 
- [x] Review & update developers documentation 
- [x] Review & update maintainers documentation 
- [x] Search for any links pointing to wiki's
- [x] Add navigation links to all pages

Update process (might do separately):
- [ ] Add missing docs:
- - [ ] Add download size entry for mod json format
- - [ ] Terrain format
- - [ ] Road format
- - [ ] River format
- - [ ] Battlefield format
- - [ ] Battle obstacles format
- - [ ] New boats format
- [ ] Merge modding docs for Bonus System into a single file
- [ ] Fix incomplete documentation for translations
- [ ] Add "Mod updates guidelines" page (1.4 only?) with list of changes in modding system and how to handle them
- [ ] Review all moddable objects and ensure that they match current schema

### CODE REVIEW

**Comment by Laserlicht** [comment]
Maybe change the whitespace in the filename to underscore? To avoid problems.

**Comment by Nordsoft91** [comment]
Vote for moving all the documentation to wiki repo:

https://github.com/vcmi/vcmi.wiki.git

1. Less sensitive to mistakes
2. Editing from browser with in-place commits
3. No restrictions in editing
4. Less review work

**Comment by IvanSavenko** [comment]
- Players documentation should be ready. It is still very basic, incomplete and poorly organized but all outdated information has been removed, formatting rechecked, and links updated.
- Modders docs - in progress
- Developer & maintainer docs - todo

**Comment by Nordsoft91** [comment]
Anticipating disagreement for every bullet so you can omit answering but still wanna give it a try and make some pros'n'cons analysis.

| Category | vcmi/vcmi.git| vcmi/vcmi.wiki.git |
|----------|--------------|--------------------|
| Entry level for new people (like non-developers from discord fro example) | Need fork, need git knowledge. Other people can give feedback about problems in documentation via issues or messages | Github account only is needed. People can fix minor problems by themselves if we encourage them |
| Online view | maybe somehow it's possible to view wiki in special tab on github, but I don't know how and this is not included in PR. So no online view for now. | Convenient online view out of the box |
| Offline view | offline view of sources for people with local copy of repo. assets are not visible offline without 3rdparty viewer. | Offline view of sources for people who additionally checkout wiki repo assets are not visible offline without 3rdparty viewer.
| Editing in browser | Editing in browser is possible from source view tab. PR for every change. Impossible to edit in browser from wiki tab | Impossible to edit from source view tab. Easy editing from wiki tab from any device (you can rely on your opinion about github poor unless you are considering yourself as only one contributor) |
| Process | Documentation update can be requested in PR (explicitly). Cost of minor change is relatively high (author time + reviewers time). Longer feedback loop for code changes | Documentation update can be requested in PR (implicitly). Cost of minor change is relatively low. Doesn't impact feedback loop for code changes |
| Code review | Review of documentation changes is embedded into code review with showing diff and ability to leave comments | Review of documentation changes is not embedded. Discussion via 3rdparty tools only |
| Versioning | Versioning is embedded into process, but not sure if github offers tool to select branch on wiki tab | Manual versioning only, like placing [VCMI-x.x] tags |
| Other | Old-style way to keep documentation in same repo. | GitHub standard to develop wiki. Functionality is being supporting and evolving by GitHub team. |

**Comment by IvanSavenko** [comment]
Sorry, what exactly are you comparing? We have:
- VCMI repository: https://github.com/vcmi/vcmi
- VCMI wiki on Github: https://github.com/vcmi/vcmi/wiki
- In theory we can create separate repository for docs in vcmi organization, but it has basically same downsides as VCMI wiki except for better online editor.

Entry level - nobody edits our current wiki other than devs. Only few edits from people without write access - krs, dydzio and one from Laserlicht. And all of them have vcmi repo set up as well. Same goes to website wiki

Online view - sorry, what? How is this different from online view?
https://github.com/IvanSavenko/vcmi/blob/documentation/docs/modders/Readme.md

Process - considering our terribly outdated docs (thank you, Warmonger, the only person who documents his changes) requiring docs update as part of PR is a good idea.

> GitHub standard to develop wiki. Functionality is being supporting and evolving by GitHub team.

Amazing editor, with syntax highlight, large edit area and preview of changes in diff form, like mediawiki. Or not...
(Screenshot made on 1440p screen by the way)

зображення

**Comment by IvanSavenko** [comment]
Also, we had this discussion month ago on Slack, with everyone in favor. Why haven't you voiced your concerns back then?

**Comment by Nordsoft91** [comment]
ah, ok

> entry level - nobody edits our current wiki other than devs

depending of what you want. If you want to change it, you need to decrease entry level. Now most of wiki materials are on website wiki, which has even higher entry level. Probably that's a root cause. Maybe not. Never know until you try.

Maybe an option would be moving all docs to GitHub wiki and monitor how it goes for few month. We always can move it inside main repo.

> How is this different from online view?

Link is different. "wiki" button is not connected to the documentation. Hard to navigate. Hard to find for people barely familiar with project. Editing process is different.

> requiring docs update as part of PR is a good idea.

maybe it's a good idea. Who stops you from requiring since it in GitHub wiki, e.g. "please update page xxx according to your changes"? Also, online editing is a big problem for PR.

> Amazing editor, with syntax highlight, large edit area and preview of changes in diff form, like mediawiki. Or not...

It's ok to consider documentation I did as "nothing" but I made all of it with GitHub editing tool. It's also allows to edit from mobile phone for example.

> we had this discussion month ago on Slack, with everyone in favor. Why haven't you voiced your concerns back then?

Maybe I had another business to do? Or maybe slack discussion is easy to miss? From which point we stopped considering pull request as a proper place to raise concerns? Anyway, very weird. And anyway, I resigned that my voice will be ignored, that's fine. My business here is small - opinion has been provided, my ask is to **consider** alternative. Do what you believes is the best.

**Comment by IvanSavenko** [comment]
Mostly done. Remaining todo:
- [ ] Recheck formatting / content of developers docs
- [ ] Add navigation links to all pages in developers docs
- [ ] Search through all internal links and make sure that they are pointing to a correct page.
- [ ] Search for any links pointing to github wiki or website wiki and update them
- [ ] Check rest of repository for links to wiki's and update them

File: docs/modders/Mod_File_Format.md
Comment (Nordsoft91) [inline docs/modders/Mod_File_Format.md:26]:
```suggestion
    // approximate size of mod, compressed by zip algorithm, in Mb
    "size" : 25,
```

Comment (IvanSavenko) [reply docs/modders/Mod_File_Format.md:26]: Do we support fractional part, e.g. `"size" : 25.3`?

**Comment by Nordsoft91** [comment]
Please, aggregate latest changes made in GitHub wiki

Also, a bit weird to see javascript in format helper

**Comment by IvanSavenko** [comment]
> Please, aggregate latest changes made in GitHub wiki

Yeah, there are also changes on website wiki.

> Also, a bit weird to see javascript in format helper

Will check Markdown docs. Website wiki uses javascript because this was the only option 10 years ago. If markdown has syntax to highlight json (and not highlight comments in red) will switch

File: docs/Readme.md
Comment (Nordsoft91) [inline docs/Readme.md:1]: will it be visible on welcome page?

Comment (IvanSavenko) [reply docs/Readme.md:1]: Welcome page?
If you mean on main page just below file list then yes.
Github docs explicitly mention that "docs/Readme.md" is one of possible locations for this file

File: docs/modders/Mod_File_Format.md
Comment (Nordsoft91) [reply docs/modders/Mod_File_Format.md:26]: yes

**Comment by Nordsoft91** [comment]
`(1.3.2) Documentation import` should be either 1.4.0 or beta branch target

**Comment by IvanSavenko** [comment]
Presumably done, will merge in a few days.

Some docs are still incomplete, but should no longer contain outdated information. If somebody can look through docs and update them where necessary - feel free to do so.

I might look into updating docs myself, but definitely not now.

**Review by Nordsoft91 — CHANGES_REQUESTED** [review]
I don't think documentation will be available from wiki tab after merge

**Comment by IvanSavenko** [comment]
> I don't think documentation will be available from wiki tab after merge into develop

No, it wont. Will add link to new docs location on wiki's

**Pull request merged**
